library/systemd/src/lib/yast2/system_service.rb

```diff
@@ -453,14 +455,9 @@ def perform_action
 
       result = send("perform_#{action}")
       register_error(:active) if result == false
-
       result
-
-      # FIXME: SystemdService#{start, stop, etc} calls to refresh! internally, so when
-      # this exception is raised we cannot distinguish if the action is failing or
-      # refresh! is failing. For SP1, refresh! should raise a new kind of exception.
-    rescue Yast::SystemctlError
-      register_error(:active)
+    rescue => e
+      register_error(:active) if e.kind_of?(Yast::SystemctlError)
```

Comment:
Hmm, I do not like it. It basically catch all errors instead of systemctl ones, is it correct?

Reply:
It seem to me, according to the expected result when  `#save` is called. If I am wrong, the other approach could be

```ruby
def save
...
rescue Yast::SystemctlError
...
rescue Yast::CouldNotRefreshUnitError
...
end
```

Reply:
I expect that `#perform_action` only catches refresh exception. The idea is to not fail if the action was performed but refresh fails, so I would do:

```
rescue Yast::SystemctlRefreshError
  true
```

Reply:
better or at least use `rescue Yast::SystemctlError, Yast::CouldNotRefreshUnitError => e` so it don't catch other problems and silent them.

Reply:
In other words, `#perform_action` should continue raising the `SystemcltError` when the action is not performed.

Reply:
But, what about to `register_error`?

Reply:
Okey, I moved `register_error` to `#save` method.